cmd/xl-v1-object.go

```diff
@@ -114,7 +146,8 @@ func (xl xlObjects) CopyObject(srcBucket, srcObject, dstBucket, dstObject string
 
 	go func() {
 		var startOffset int64 // Read the whole file.
-		if gerr := xl.GetObject(srcBucket, srcObject, startOffset, length, pipeWriter); gerr != nil {
+		srcEtag := xlMeta.ToObjectInfo(srcBucket, srcObject).ETag
+		if gerr := xl.getObject(srcBucket, srcObject, startOffset, length, pipeWriter, srcEtag); gerr != nil {
```

Comment:
you can pass `""` for etag here as the validation is already done above

Reply:
done.